app/workers/functions.php

```diff
@@ -412,48 +371,61 @@ public function execute(string $trigger, string $projectId, string $executionId,
-                (empty($swap) ? "" : (" --memory-swap=".$swap."m")).
-                " --name={$container}".
-                " --label appwrite-type=function".
-                " --label appwrite-created={$executionTime}".
-                " --volume {$tagPathTargetDir}:/tmp:rw".
-                " --workdir /usr/local/src".
-                " ".\implode(" ", $vars).
-                " {$runtime['image']}".
-                " tail -f /dev/null"
-            , '', $stdout, $stderr, 30);
-
-            if($exitCode !== 0) {
-                throw new Exception('Failed to create function environment: '.$stderr);
-            }
 
-            $exitCodeUntar = Console::execute("docker exec ".
-                $container.
-                " sh -c 'mv /tmp/code.tar.gz /usr/local/src/code.tar.gz && tar -zxf /usr/local/src/code.tar.gz --strip 1 && rm /usr/local/src/code.tar.gz'"
-                , '', $stdout, $stderr, 60);
+            $orchestration->setCpus($cpus);
+            $orchestration->setMemory($memory);
+            $orchestration->setSwap($swap);
 
-            if($exitCodeUntar !== 0) {
-                throw new Exception('Failed to extract tar: '.$stderr);
+            \array_walk($vars, function (string &$value) {
+                $value = strval($value);
+            });
+
+            $id = $orchestration->run(
+                $runtime['image'],
+                $container,
+                "",
+                array(
+                    'tail',
+                    '-f',
+                    '/dev/null'
+                ),
+                "/usr/local/src",
+                array(),
+                $vars,
+                $tagPathTargetDir,
+                array(
+                    "appwrite-type" => "function",
+                    "appwrite-created" => "{$executionTime}"
+                ));
+
+            $tarStdout = '';
+            $tarStderr = '';
+
+            $untarSuccess = $orchestration->execute($container, 
+                array(
+                    'sh',
+                    '-c',
+                    'mv /tmp/code.tar.gz /usr/local/src/code.tar.gz && tar -zxf /usr/local/src/code.tar.gz --strip 1 && rm /usr/local/src/code.tar.gz'), 
+                $tarStdout, 
+                $tarStderr,
+                $vars,
+                60);
+
+            if (!$untarSuccess) {
+                throw new Exception('Error untarring code');
             }
 
             $executionEnd = \microtime(true);
 
-            $list[$container] = [
-                'name' => $container,
-                'online' => true,
-                'status' => 'Up',
-                'labels' => [
+            $list[$container] = new Container($container, 
+                $id, 
+                "Up", 
+                array(
                     'appwrite-type' => 'function',
-                    'appwrite-created' => $executionTime,
-                ],
-            ];
+                    'appwrite-created' => "{$executionTime}"
```

Comment:
```suggestion
                    'appwrite-created' => $executionTime
```

No need for the template literal.